packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx

```diff
@@ -0,0 +1,117 @@
+             * and appends them to the newParams interface.
+             * 
+             * @param {string} queryParam 
+             * @param {string} prop 
+             */
+            function handleQueryParams(queryParam) {
+                if (currentParams.has(queryParam)) {
+                    const queryParameter = currentParams.get(queryParam);
+                    newParams.append(queryParam, queryParameter);
+                }
+            }
+
+            /**
+             * Function that handles the presence of query parameters and props
+             * and appends them to the newParams interface.
+             * 
+             * @param {string} queryParam 
+             * @param {string} prop 
+             */
+            function handleQueryParamsAndProps(queryParam, prop) {
+                if (currentParams.has(queryParam)) {
+                    const queryParameter = currentParams.get(queryParam);
+                    newParams.append(queryParam, queryParameter);
+                } else if (prop) {
+                    if (prop === primaryColorProp) {
+                        newParams.append(queryParam, primaryColorProp.replace("#", ""))
+                    } else {
+                        newParams.append(queryParam, prop)
+                    }
+                }
+            }
+
+            // Handle query parameters for the gmApiKey and mapboxAccessToken
+            handleQueryParams('gmApiKey');
+            handleQueryParams('mapboxAccessToken');
+
+            // Handle query parameters and props for apiKey, primaryColor, logo 
+            handleQueryParamsAndProps('apiKey', apiKeyProp);
+            handleQueryParamsAndProps('primaryColor', primaryColorProp);
+            handleQueryParamsAndProps('logo', logoProp);
+
+            // Get the string with all the final query parameters
+            const finalParams = newParams.toString()
+
+            // Construct the QR code URL
+            let QRCodeURL = `${currentUrl}/?${finalParams}&directionsFrom=${directionsFrom.id}&directionsTo=${directionsTo.id}`
+
+            const options = {
+                errorCorrectionLevel: 'L',
+                margin: 0,
+                width: '225'
+            };
+
+            QRCode.toDataURL(QRCodeURL, options)
+                .then((dataUrl) => {
+                    document.getElementById('qr').src = dataUrl;
+                }).catch(() => {
+                    console.log('error')
+                });
+        }
+
+    }, [directionsFrom, directionsTo])
+
+
+    function closeDialog() {
+        setShowQRCodeDialog(false);
+    }
+    return (<>
+        <div className="background"></div>
+        <div className="qr-code">
+            <img id='qr' alt="QR Code" className="qr-code__image" />
+            <p>Scan the QR code to see the route on your phone</p>
+            <button className="qr-code__button" style={{ background: primaryColorProp }} onClick={() => closeDialog()}>DONE</button>
```

Comment:
Use CSS to transform the text to uppercase.

Reply:
✅